### DISCUSSION TITLE
1D MESH subdomain/block creation

### DISCUSSION THREAD

**Discussion by ananthanarasimhanj** [idaholab/moose#24791] [Q&A Getting Started] [answered]
I want to perform 1D heat conduction, and it involves two materials. I am now trying to create a mesh, and assign them as made of 2 blocks, to define material properties, and BC. I am not finding relevant guidance to assigning "type" for the 1D subdomains to create. Below is the incomplete code for MESH. 
```
[Mesh]
  [generated]
    type = GeneratedMeshGenerator
    dim = 1
    nx = 10
    xmax = 1
  []

  [air_block]
    type = 
    input = generated
    block_id = 1
    left = 0
    right = 0.5
  []

  [water_block]
    type = 
    input = generated
    block_id = 2
    left = 0.5
    right = 1
  []

[]
```

**Comment by lindsayad** [comment]
You probably want `type = SubdomainBoundingBoxGenerator`

**Reply by ananthanarasimhanj** [reply]
I tried that and it shows error. I am attaching the input file, and the error too. 

[Mesh]
  [generated]
    type = GeneratedMeshGenerator
    dim = 1
    nx = 10
    xmax = 1
    block_id = 0
  []

  [air_block]
    type = SubdomainBoundingBoxGenerator
    input = generated
    block_id = 1
    bottom_left = '0 0 0'
    top_right = '0.5 0 0'
  []

  [water_block]
    type = SubdomainBoundingBoxGenerator
    input = generated
    block_id = 2
    bottom_left = '0.5 0 0'
    top_right = '1 0 0'
  []

[]

[Variables]
  [T]
    initial_condition = 300.0
  []
[]

[Kernels]
  [heat_conduction]
    type = HeatConduction
    variable = T
  []

  [time_derivative]
    type = HeatConductionTimeDerivative
    variable = T
  []

[]

[Materials]
  [thermal1]
    type = HeatConductionMaterial
    thermal_conductivity = 0.02587 # 20 degree Celcius, W/mK
    specific_heat = 1006 # Cp, 20 degree Celcius, J/kgK
    block = 1
  []

  [thermal2]
    type = HeatConductionMaterial
    thermal_conductivity = 0.598 # 20 degree Celcius, W/mK
    specific_heat = 4150 # Cv, 20 degree Celcius, J/kgK
    block = 2
  []


  [density1]
    type = GenericConstantMaterial
    prop_names = 'density'
    prop_values = 1.204 # 20 degree Celcius kg/m3
    block = 1
  []

  [density2]
    type = GenericConstantMaterial
    prop_names = 'density'
    prop_values = 998.19 # 20 degree Celcius kg/m3
    block = 2
  []


[]

[BCs]
  [t_left]
    type = DirichletBC
    variable = T
    value = 373.15
    boundary = 'left'
    block = 1
  []

  [t_right]
    type = FunctionNeumannBC
    variable = T
    function = 0
    boundary = 'right'
    block = 2
  []


 
[]

[Executioner]
  type = Transient
  end_time = 4
  dt = 1e-3
[]

[VectorPostprocessors]
  [t_sampler]
    type = LineValueSampler
    variable = T
    start_point = '0 0 0'
    end_point = '1 0 0'
    num_points = 20
    sort_by = x
  []
[]

[Outputs]
  exodus = true
  [csv]
    type = CSV
    file_base = hc3_out
    execute_on = final
  []
[]



*** ERROR ***
Your MeshGenerator tree contains multiple possible generator outputs :
"water_block and one or more of the following from an independent set: "air_block"

This may be due to a missing dependency or may be intentional. Please either
- check that all the mesh generators are connected as a tree and culminate in a single final mesh. Having one wrong 'input=mg' parameter is the most common error
- add additional dependencies to remove the ambiguity if you are using a user-built MeshGenerator
- if you intend to execute a subset of the defined generators (uncommon), select the final MeshGenerator in the [Mesh] block with the "final_generator" parameter.

**Reply by GiudGiud** [reply, marked as answer]
try
```
[water_block]
type = SubdomainBoundingBoxGenerator
input = air_block
block_id = 2
bottom_left = '0.5 0 0'
top_right = '1 0 0'
[]
```

**Reply by ananthanarasimhanj** [reply]
If the input is air_block, then the coordinates I have mentioned for water_block are outside air_block. Is that fine? I tried this, and I am getting the following error. 

*** ERROR ***
/home/ananth/projects/hc/hc1/hc3.i:12: unused parameter 'Mesh/generated/block_id'
/home/ananth/projects/hc/hc1/hc3.i:90: unused parameter 'BCs/t_left/block'
/home/ananth/projects/hc/hc1/hc3.i:98: unused parameter 'BCs/t_right/block'


below snapshot shows what I want to do. 

![image](https://github.com/idaholab/moose/assets/113285250/d7eb83ac-a419-454f-903a-d61ba625638f)

**Reply by GiudGiud** [reply]
These unused parameter errors must be addressed.
Boundary conditioons apply on boundaries not blocks. 

And one of the mesh parameters is wrong too. Please check the object documentation

**Reply by ananthanarasimhanj** [reply]
Sure. Thanks for the help.

**Reply by ananthanarasimhanj** [reply]
I tried below code, and it seems working. 
```
[Mesh]
  [air_block]
    type = GeneratedMeshGenerator
    dim = 1
    nx = 10
    xmax = 1
  []

  [water_block]
    type = SubdomainBoundingBoxGenerator
    input = air_block
    block_id = 1
    bottom_left = '0.5 0 0'
    top_right = '1 0 0'
  []
[]

[Variables]
  [T]
    initial_condition = 300.0
  []
[]

[Kernels]
  [heat_conduction]
    type = HeatConduction
    variable = T
  []

  [time_derivative]
    type = HeatConductionTimeDerivative
    variable = T
  []

[]

[Materials]
  [thermal1]
    type = HeatConductionMaterial
    thermal_conductivity = 0.02587 # 20 degree Celcius, W/mK
    specific_heat = 1006 # Cp, 20 degree Celcius, J/kgK
    block = 0
  []

  [thermal2]
    type = HeatConductionMaterial
    thermal_conductivity = 0.598 # 20 degree Celcius, W/mK
    specific_heat = 4150 # Cv, 20 degree Celcius, J/kgK
    block = 1
  []


  [density1]
    type = GenericConstantMaterial
    prop_names = 'density'
    prop_values = 1.204 # 20 degree Celcius kg/m3
    block = 0
  []

  [density2]
    type = GenericConstantMaterial
    prop_names = 'density'
    prop_values = 998.19 # 20 degree Celcius kg/m3
    block = 1
  []


[]

[BCs]
  [t_left]
    type = DirichletBC
    variable = T
    value = 373.15
    boundary = 'left'
  []

  [t_right]
    type = FunctionNeumannBC
    variable = T
    function = 0
    boundary = 'right'
  []


 
[]

[Executioner]
  type = Transient
  end_time = 4
  dt = 1e-3
[]

[VectorPostprocessors]
  [t_sampler]
    type = LineValueSampler
    variable = T
    start_point = '0 0 0'
    end_point = '1 0 0'
    num_points = 20
    sort_by = x
  []
[]

[Outputs]
  exodus = true
  [csv]
    type = CSV
    file_base = hc3_out
    execute_on = final
  []
[]
```
